hypnos/src/clock.c

```diff
@@ -70,32 +72,47 @@ void clock_increment_local_time()
 
 void clock_print_time()
 {
-	char wday[15];
-	char mon[15];
+	char wday[10];
+	char mon[10];
 
 	if (sscanf(clock_get_local_time(), "%s %s", wday, mon) != 2) {
 		LOG_ERR("Failed to print time!");
 	}
-	LOG_INF("%s %d %s\n", log_strdup(wday), localtime(&local_time)->tm_mday,
+	LOG_INF("%s %d %s", log_strdup(wday), localtime(&local_time)->tm_mday,
 		log_strdup(mon));
-	LOG_INF("%d:%d:%d", localtime(&local_time)->tm_hour,
-	       localtime(&local_time)->tm_min,
-	       localtime(&local_time)->tm_sec);
-	sprintf(clock_label_str, "%d:%d:%d", localtime(&local_time)->tm_hour,
-		localtime(&local_time)->tm_min,
-		localtime(&local_time)->tm_sec);
+	LOG_INF("%02d:%02d", localtime(&local_time)->tm_hour,
+		localtime(&local_time)->tm_min);
+	sprintf(clock_label_str, "%02d:%02d", localtime(&local_time)->tm_hour,
+		localtime(&local_time)->tm_min);
+	sprintf(date_label_str, "%s %d %s", wday,
+		localtime(&local_time)->tm_mday, mon);
 
+	/* Make the hours:minutes separator blink to represent seconds */
+	if (local_time % 2) {
+		clock_label_str[2] = ' ';
+	}
 	lv_label_set_text(clock_label, clock_label_str);
+	lv_label_set_text(date_label, date_label_str);
 }
 
 void clock_gfx_init()
 {
-	sprintf(clock_label_str, "%d:%d:%d", localtime(&local_time)->tm_hour,
-	       localtime(&local_time)->tm_min,
-	       localtime(&local_time)->tm_sec);
+	char wday[15];
```

Comment:
The char of wday and mon is of size 10 in clock_print_time(). Can size 10 work here as well?

Reply:
Actually, I realised that wday and mon are abbreviations (three chars), so I can reduce even more.

Reply:
I changed it to `4` in two places it that allowed me to decrease `date_label_str` to size `32`, thanks!